strax/processing/peak_merging.py

```diff
@@ -159,6 +173,10 @@ def add_lone_hits(peaks, lone_hits, to_pe):
         lh_area = lh_i['area'] * to_pe[lh_i['channel']]
         p['area'] += lh_area
 
-        # Add lone hit as delta pulse to waveform:
+        # Add lone hit as delta pulse to waveforms:
         index = (p['time'] - lh_i['time'])//p['dt']
         p['data'][index] += lh_area
+
+        if n_top_channels > 0:
+            if lh_i['channel'] < n_top_channels:
+                p['data_top'][index] += lh_area
```

Comment:
@terliuk this is the line which adds the lone hit information to the top waveform.